src/VideoRemise/VideoRemise/VideoChannel.cs

```diff
@@ -171,59 +196,198 @@ internal async Task ShutdownAsync()
         {
+            currentReplay = Math.Min(currentReplay + 1, actions.Count - 1);
+            PlayFromFile();
+            mainPage.CurrentMode = Mode.Replaying;
+        }
 
+        internal void Live()
+        {
+            captureElement.Visibility = Visibility.Visible;
+            mediaPlayerElement.Visibility = Visibility.Collapsed;
+            showingLive = true;
+            playing = false;
         }
 
         internal async Task<IAsyncAction> StartRecording(string fileBaseName)
         {
-            //currentRecordingStream = new InMemoryRandomAccessStream();
+            baseName = fileBaseName;
+            return await StartPhrase();
+        }
+
+        internal async Task<IAsyncAction> StartPhrase()
+        {
             var myVideos = await StorageLibrary.GetLibraryAsync(Windows.Storage.KnownLibraryId.Videos);
             var fencingVideos = await myVideos.SaveFolder.CreateFolderAsync(SaveSubfolderName,
                 CreationCollisionOption.OpenIfExists);
-            var file = await fencingVideos.CreateFileAsync($"{fileBaseName}-{ChannelName}.mp4", 
+            currentFile = await fencingVideos.CreateFileAsync($"{baseName}-{ChannelName}.mp4",
                 CreationCollisionOption.GenerateUniqueName);
-            mediaRecording = await currentCapture.PrepareLowLagRecordToStorageFileAsync(MediaEncodingProfile.CreateMp4(VideoEncodingQuality.Auto), file);
-            //mediaRecording = await currentCapture.PrepareLowLagRecordToStreamAsync(MediaEncodingProfile.CreateMp4(VideoEncodingQuality.Auto), currentRecordingStream);
+            mediaRecording = await currentCapture.PrepareLowLagRecordToStorageFileAsync(MediaEncodingProfile.CreateMp4(VideoEncodingQuality.Auto), currentFile);
             isRecording = true;
             return mediaRecording.StartAsync();
         }
 
+        internal async Task EndPhrase(int length, TriggerType triggerType)
+        {
+            if (!isRecording)
+            {
+                return;
+            }
+
+            await StopRecording();
+            PhraseRecording pr = new PhraseRecording(currentFile, length, triggerType);
+            actions.Add(pr);
+        }
+
         internal async Task<IAsyncAction> StopRecording()
         {
             isRecording = false;
             await mediaRecording.StopAsync();
             return mediaRecording.FinishAsync();
         }
 
-        internal void StartPlayback()
+        internal async void Trigger(int length, TriggerType triggerType)
         {
-            captureElement.Visibility = Visibility.Collapsed;
-            mediaPlayerElement.Visibility = Visibility.Visible;
-            showingLive = false;
-            mediaPlayerElement.Source = MediaSource.CreateFromStream(currentRecordingStream, MediaEncodingProfile.CreateMp4(VideoEncodingQuality.Auto).ToString());
-            mediaPlayerElement.MediaPlayer.Play();
+            if (!isRecording)
+            {
+                return;
```

Comment:
We might want to do something to alert the user that they're not recording in this situation -- I'm not sure what. I'll open an issue for it.